src/components/side-panel/common/types.ts

```diff
@@ -40,31 +36,31 @@ export type StyledDatasetTitleProps = {
 
 export type DatasetTitleProps = {
   dataset: KeplerTable;
-  showDatasetTable: Function;
   showDeleteDataset: boolean;
   onTitleClick?: () => void;
-  updateTableColor?: (d: string, c?: RGBColor) => void;
-  removeDataset?: (d: string) => void;
+  showDatasetTable?: typeof VisStateActions.showDatasetTable;
+  updateTableColor: typeof VisStateActions.updateTableColor;
+  removeDataset?: typeof openDeleteModal;
 };
 
 export type SourceDataCatalogProps = {
-  datasets: Datasets;
-  showDatasetTable: Function;
-  showDeleteDataset: boolean;
+  datasets: KeplerTable[];
+  showDeleteDataset?: boolean;
   onTitleClick?: () => void;
-  updateTableColor?: (d: string, c?: RGBColor) => void;
-  removeDataset?: (d: string) => void;
+  showDatasetTable?: typeof VisStateActions.showDatasetTable;
+  updateTableColor: typeof VisStateActions.updateTableColor;
+  removeDataset?: typeof openDeleteModal;
 };
 
 export type DatasetItemProps = {
   value: KeplerTable;
 };
 
 export type SourceDataSelectorProps = {
-  dataId: string;
-  datasets: Datasets;
+  dataId: string | string[];
+  datasets: KeplerTable[];
```

Comment:
should not change `Datasets` to `KeplerTable[]`

Reply:
fixed